docs/shard.yml.schema.json

```diff
@@ -0,0 +1,262 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "shard.yml",
+  "description": "Metadata for projects managed by Shards",
+  "type": "object",
+  "properties": {
+    "name": {
+      "type": "string",
+      "title": "project name",
+      "description": "The name of the project",
+      "maxLength": 50,
+      "pattern": "^(?!.*__|.*--|.*crystal|[0-9_-])[a-z0-9_-]+(?<![_-])$"
+    },
+    "version": {
+      "type": "string",
+      "title": "project version",
+      "description": "The version number of the project",
+      "pattern": "^(?!.*\\.\\.|.*--)[0-9a-zA-Z.-]+$"
+    },
+    "authors": {
+      "type": "array",
+      "title": "authors",
+      "description": "A list of authors, along with their contact email",
+      "items": {
+        "type": "string",
+        "title": "author",
+        "description": "An authors name and contact email"
+      }
+    },
+    "crystal": {
+      "type": "string",
+      "title": "crystal version",
+      "description": "A restriction to indicate what Crystal versions are supported"
+    },
+    "dependencies": {
+      "type": "object",
+      "title": "dependencies",
+      "description": "A list of required dependencies\nhttps://github.com/crystal-lang/shards/blob/master/docs/shard.yml.adoc#dependency-attributes",
```

Comment:
**thought (non blocking):** I'm wondering if we could host a HTML rendering of the `shard.yml` spec at a stable place under https://crystal-lang.org and link to that page instead of a github file that may be moved :thinking: 

poke @straight-shoota

Reply:
Good point, we should certainly do that. Would fit right next to https://crystal-lang.org/reference/1.11/man/shards
Not sure why we're missing it there 🤔
Anyway, we can merge this as is and update the URL later.

Reply:
Ah, seems like I had this idea a long time ago already: https://github.com/crystal-lang/crystal-book/issues/476
But we haven't managed to make the format compatible with HTML rendering: https://github.com/crystal-lang/shards/pull/488